packages/server/src/socket/service.ts

```diff
@@ -0,0 +1,88 @@
+import { uuid } from '@botpress/messaging-base'
+import { Socket } from 'socket.io'
+import { Service } from '../base/service'
+import { ServerCache } from '../caching/cache'
+import { CachingService } from '../caching/service'
+import { UserService } from '../users/service'
+
+export class SocketService extends Service {
+  private sockets: { [socketId: string]: SocketState | undefined } = {}
+  private cache!: ServerCache<string, SocketState>
+
+  private socketsByUserId: { [userId: string]: Socket[] | undefined } = {}
+  private cacheByUserId!: ServerCache<uuid, Socket[]>
+
+  constructor(private cachingService: CachingService, private userService: UserService) {
+    super()
+  }
+
+  async setup() {
+    this.cache = await this.cachingService.newServerCache('cache_socket_states')
+    this.cacheByUserId = await this.cachingService.newServerCache('cache_sockets_by_user_id')
+  }
+
+  public create(socket: Socket) {
+    this.sockets[socket.id] = {}
+  }
+
+  public delete(socket: Socket) {
+    const state = this.sockets[socket.id]!
```

Comment:
Add a guard that checks if the socket.id is in the sockets dictionary and return if not.

Reply:
Exceptions for this function are already catched from the caller. I don't think the function should silently do nothing like that. If we call deleted on a socket that doesn't exist or as already been deleted then that's wrong usage and we should be informed about it.